# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/CanvasClient.cpp
# Commit: b15d91e64a25
# Full Hash: b15d91e64a2589d47ad023d61dbd45135669eb60
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 71ad2ed8e9ba -r b15d91e64a25 gfx/layers/client/CanvasClient.cpp
--- a/gfx/layers/client/CanvasClient.cpp	Wed Jun 10 22:21:07 2020 +0000
+++ b/gfx/layers/client/CanvasClient.cpp	Wed Jun 10 22:21:09 2020 +0000
@@ -31,6 +31,12 @@
 void CanvasClient::UseTexture(TextureClient* const aTexture) {
   MOZ_ASSERT(aTexture);
 
+  const auto isClientNonPremult =
+      bool(mTextureFlags & TextureFlags::NON_PREMULTIPLIED);
+  const auto isTextureNonPremult =
+      bool(aTexture->GetFlags() & TextureFlags::NON_PREMULTIPLIED);
+  MOZ_ALWAYS_TRUE(isTextureNonPremult == isClientNonPremult);
+
   bool changed = false;
 
   if (aTexture != mFrontBuffer) {
